Adds nosql schema structure for the assignment

shirishwebonise opened the pull request.

eliza-abraham committed:
Merge pull request #1 from shirishwebonise/staging

Adds descriptive files to the Assignment repository

Commit:
Merge pull request #4 from shirishwebonise/develop

Develop

Commit:
Merge pull request #5 from shirishwebonise/testing

Testing

Commit:
Merge pull request #6 from shirishwebonise/master

Merge pull request #1 from shirishwebonise/staging

Commit:
Merge pull request #8 from shirishwebonise/staging

Staging

shirishparsekar committed:
Add nosql assignment
  added:
	nosql schema structure

shirishparsekar committed:
Correct format

shirishparsekar committed:
Correct format
	correction in structure format
	near  profile.about_me (multiline string)

shirishparsekar committed:
Add demo operations
	-add account
	-add profile info

eliza-abraham commented on nosql/assignment.nosql.schema.txt:
```diff
@@ -0,0 +1,84 @@
+				"i am this,"+
+				"i am this"
+	},
+	posts: [
```
posts should be a separate collection 
- how would you fetch all the posts, or latest posts? The computation to fetch posts from multiple sub documents to sort, or list would be very high. 
- Similarly while editors have access to certain posts, you would need to check against each sub document of the users. .

eliza-abraham commented on nosql/assignment.nosql.schema.txt:
```diff
@@ -0,0 +1,84 @@
+		{
+			heading: "",
+			body: "",
+			datetime: "",
```
should be specific about what kind of timestamp it is
eg: created_at, updated_at, posted_at, published_at.

eliza-abraham commented on nosql/assignment.nosql.schema.txt:
```diff
@@ -0,0 +1,84 @@
+			],
+			comments: [
+				{
+					datetime: "",
```
be specific about what kind of timestamp it is

eliza-abraham commented on nosql/assignment.nosql.schema.txt:
```diff
@@ -0,0 +1,84 @@
+Collection - Users
+
+{
+	account: {
```
Also specify the default _id which is added by mongo

eliza-abraham commented on nosql/assignment.nosql.schema.txt:
```diff
@@ -0,0 +1,84 @@
+				"",
+				""
+			],
+			comments: [
```
Make a practice to add primary keys to each document in an array of sub documents

eliza-abraham commented on nosql/nosql.functions:
```diff
@@ -0,0 +1,120 @@
+
+<< updating profile information >>
+
+db.users.find({ "account.username" : "shirish.sp"})
```
Why find over findOne?

eliza-abraham commented on nosql/nosql.functions:
```diff
@@ -0,0 +1,120 @@
+			}
+		}
+	}
+)
```
always add new line at eof! 
PS: Set it in your editor.

eliza-abraham commented on nosql/assignment.nosql.schema.txt:
```diff
@@ -0,0 +1,84 @@
+	account: {
+		username: "shirish.sp",
+		password: "**********",
+		roles: [
```
- User will have only one role, user cannot be an admin and super admin at the same time

eliza-abraham replied:
Use Case: 
- Admin would have access only to a subset of users. And he would be able to manage only those users, and those posts of those users assigned to him 
- Similarly, an editor can edit only those posts which are assigned to him

eliza-abraham commented on nosql/assignment.nosql.schema.txt:
```diff
@@ -0,0 +1,84 @@
+
+{
+	account: {
+		username: "shirish.sp",
```
It;s fine to have username as a primary/unique key, but for reference its always better to use the collection's _id, since it is indexed. Faster retrieval.

shirishparsekar committed:
Fix db structure

shirishparsekar committed:
Add queries with fixed structure

shirishparsekar committed:
Update queries with review answers

Commit:
Consider review suggestions
  - reduced nesting at    comments->posted_by->{}

Commit:
Consider better ways for queries
  - display number of comments on a specific post
  - display posts with number of comments greater than 3

eliza-abraham commented on nosql/nosql.functions:
```diff
@@ -0,0 +1,406 @@
+
+--------------------------------------------------------------------------
+
+db.posts.find(
```
When you are fetching a single post specially with _id, use findOne() over find()
